starknet_devnet/blueprints/rpc/misc.py

```diff
@@ -63,15 +63,18 @@ async def syncing() -> Union[dict, bool]:
     return False
 
 
-# pylint: disable=too-many-arguments
-async def get_events(
-    from_block: BlockId,
-    to_block: BlockId,
-    chunk_size: int,
-    address: Address = None,
-    keys: List[Address] = None,
-    continuation_token: str = "0",
-) -> str:
+def try_parse_from_filter(filter_data, name):
+    """
+    Try to parse data from filter data. Return None if there is no data for a given name.
+    """
+    try:
+        return filter_data[name]
+    except KeyError:
+        return None
+
+# pylint: disable=redefined-builtin
+# filter name is determined by current RPC implementation and starknet specification
```

Comment:
Why not just call the property `filter_data`, that way there is no need to redefine the builtin.

Reply:
I tried that but it will lead to problems in this code:

```

rpc = Blueprint("rpc", __name__, url_prefix="/rpc")


@rpc.route("", methods=["POST"])
async def base_route():
    """
    Base route for RPC calls
    """

    message_id = None
    try:
        method, params, message_id = parse_body(request.json)
        result = await (
            method(*params) if isinstance(params, list) else method(**params)
        )
        if inspect.iscoroutinefunction(result):
            result = await result
    except TypeError as type_error:
        return rpc_error(message_id=message_id, code=22, message=str(type_error))
    except RpcError as error:
        return rpc_error(message_id=message_id, code=error.code, message=error.message)

    return rpc_response(message_id=message_id, content=result)


def parse_body(body: dict) -> Tuple[Callable, Union[List, dict], int]:
    """
    Parse rpc call body to function name, params and message id
    """
    try:
        method_name = body["method"].replace("starknet_", "")
        params: Union[List, dict] = body.get("params") or {}
        message_id = body["id"]
    except RuntimeError as error:
        raise RpcError(
            code=RpcErrorCode.INVALID_REQUEST.value, message="Invalid request"
        ) from error

    if method_name not in methods:
        raise RpcError(
            code=RpcErrorCode.METHOD_NOT_FOUND.value, message="Method not found"
        )

    if not isinstance(params, (List, Dict)):
        raise RpcError(code=RpcErrorCode.INVALID_PARAMS.value, message="Invalid params")

    return methods[method_name], params, message_id
```

Reply:
Ah right, ok! We can leave as it is with the suppression. Marking as resolved.